libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/permissions/DocumentPermissions.java

```diff
@@ -125,12 +123,7 @@ protected boolean isClearingAdminOfOwnGroup() {
                 .collect(Collectors.toList());
     }
 
-    public boolean isAllowedToDownload(AttachmentContent attachment){
-        return isAllowedToDownload(attachment.getId());
-    }
-
-    public boolean isAllowedToDownload(String attachmentContentId){
-        return nullToEmptySet(getAttachmentContentIds()).contains(attachmentContentId) &&
```

Comment:
I think that is for the "one is only allowed to download an attachment if it is in the list of attachments of something AND if one is allowed to read the something"

Without that check one is able to download arbitrary attachments in the context of something to which one has read access.

Reply:
> > - Why do we need to verify the attachment context and permissions for components and releases. [...]
> 
> to be future proof and to have a generic and uniform implementation.

If we provide some endpoint to download atatchments (e.g. of an release) without context, then one could use that endpoint to again download arbitrary attachments.